@here/olp-sdk-dataservice-read/lib/client/VersionedLayerClient.ts

```diff
@@ -235,16 +240,21 @@ export class VersionedLayerClient {
         );
         const latestVersion =
             version || (await this.getLatestVersion(billingTag));
-        const partitions = await QueryApi.getPartitionsById(
+        const partitionsList = await QueryApi.getPartitionsById(
             queryRequestBilder,
             {
                 version: `${latestVersion}`,
                 layerId: this.layerId,
                 partition: [partitionId],
                 billingTag
             }
-        );
-        const partition = partitions.partitions.find(element => {
+        ).catch(async error => Promise.reject(error));
+
+        if (!partitionsList || !partitionsList.partitions) {
```

Comment:
Please update return model in data-service-api and so on..

Reply:
Updated